web-app/js/portal/ui/openlayers/SpatialConstraintMap.js

```diff
@@ -1,6 +1,9 @@
 OpenLayers.SpatialConstraintMap = OpenLayers.Class(OpenLayers.Map, {
 
     initialize: function(div, options) {
+        if(div) {
```

Comment:
This is a bit weird here. I'm not sure why `initialise` is taking two arguments? Only `options` or null are supplied
```
OpenLayers.SpatialConstraintMap = OpenLayers.Class(OpenLayers.Map, {

    initialize: function(options) {

        if(options) {
            this.navigationControl = options.navigationControl;
        }
        ......
```

Reply:
Yeah - looks weird, but it's old weird stuff - not new weird stuff  :)

Reply:
Yep old and now time to fix

Reply:
Nope - it has two arguments because it is over-riding the base class initialize method.

Reply:
Erk. Because of this, these lines aren't needed
```
if(div) {
            this.navigationControl = div.navigationControl;
}
```
 `this.navigationControl` gets created by the `OpenLayers.Map.prototype.initialize.apply(this, arguments);`

Reply:
it is needed - because that's how i put a handle to the navigation control in here - so that i can check it's state in the enableControl & disableControl methods.